spec/site_github_munger_spec.rb

```diff
@@ -32,6 +32,22 @@
       end
     end
 
+    context "with site.baseurl set to ''" do
+      let(:user_config) { { "baseurl" => "" } }
+
+      it "doesn't mangle site.baseurl" do
+        expect(site.config["baseurl"]).to eql("")
+      end
+    end
+
+    context "with site.baseurl set to '/'" do
+      let(:user_config) { { "baseurl" => "/" } }
+
+      it "mangles site.url" do
+        expect(site.config["baseurl"]).to eql("/github-metadata")
```

Comment:
That was the original logic. 

* Scenario 1: Users incorrectly set their baseurl as `/`
* Scenario 2: A sophisticated user is proxying their site and wants their baseurl to be `/` despite it being a project page

I'd lean towards mangling, as in scenario 1, the user needs our help more than the user in scenario 2.